civis/ml/_model.py

```diff
@@ -268,6 +259,122 @@ def _show_civisml_warnings(warn_list):
             warnings.warn(warn_str, RuntimeWarning)
 
 
+def _get_job_type_version(alias):
+    """Derive the job type and version from the given alias.
+
+    Parameters
+    ----------
+    alias : str
+        CivisML alias
+
+    Returns
+    -------
+    str
+        Job type, one of {training, prediction, registration}.
+    str
+        CivisML version, e.g., "v2.2".
+    """
+    # A version-less alias for production, e.g., "civis-civisml-training"
+    match_production = re.search(r'\Acivis-civisml-(\w+)\Z', alias)
+    # A versioned alias, e.g., "civis-civisml-training-v2-3"
+    match_v = re.search(r'\Acivis-civisml-(\w+)-v(\d+)-(\d+)\Z', alias)
+    # A special-version alias, e.g., "civis-civisml-training-dev"
+    match_special = re.search(r'\Acivis-civisml-(\w+)-(\S+[^-])\Z', alias)
+
+    if match_production:
+        job_type = match_production.group(1)
+        version = None
+    elif match_v:
+        job_type = match_v.group(1)
+        version = 'v%s.%s' % match_v.group(2, 3)
+    elif match_special:
+        job_type = match_special.group(1)
+        version = match_special.group(2)
+    else:
+        msg = '"%r" does not look like a CivisML alias'
+        raise ValueError(msg % alias)
+
+    return job_type, version
+
+
+def _get_template_ids_all_versions(client):
+    """Get templates IDs for all accessible CivisML versions.
+
+    Parameters
+    ----------
+    client : APIClient
+        Civis API client object
+
+    Returns
+    -------
+    Dict[str, Dict[str, int]]
+        Mapping between versions (e.g., "v2.2") and template IDs for the given
+        version (e.g., {'training': 1, 'prediction': 2, 'registration': 3}).
+    """
+    template_alias_objects = client.aliases.list(
+        object_type='template_script', iterator=True
+    )
+    civisml_template_alias_objects = find(
+        template_alias_objects,
+        alias=lambda alias: alias.startswith('civis-civisml-')
+    )
+    ids = collections.defaultdict(
+        lambda: {'training': None, 'prediction': None, 'registration': None}
```

Comment:
I considered some sort of `collections.namedtuple` to represent `{'training': None, 'prediction': None, 'registration': None}`, but decided that the dictionary syntax is clearer for readable and maintainability.